File: calculator.py
def hr_to_btc(hr, d, r):
    ''' int, int, int -> float 
    Calculates the expected mine assets a according to a hash rate h
    '''
    ghr = d * 2**32  / ( t )  # EH/s = H * 10^18  /s
    a = hr / ghr * r

    return a

def btc_to_hr(a, d, r):
    ''' int, int, int -> float 
    Calculates the expected mine assets a according to a hash rate h
    '''
    ghr = d * 2**32 / (t)
    hr = (a/r) * ghr

    return hr

def btc_to_dollar(btc):
    dollar_one_btc = 8938.65
    return dollar_one_btc * btc
    

if __name__ == '__main__':
    d = 14776367535688  # bits
    r = 12.5    # bitcoin
    t = 84600
    h = 2367890000 

File: calculator.js
var calcul = function(){
    var bool;
    document.getElementById("");
    return bool
}

var hr_to_btc  = function (hr, d, bool){
    var r = 12.5;  
    var ghr = d * Math.pow(2,32) / t;
    var a = hr/ghr*r;

    return a;

}

var btc_to_hr = function(a, d){
    var r = 12.5;  
    var ghr = d * Math.pow(2,32) /t;
    var hr = (a/r)* ghr;

    return hr

}

var  btc_to_dollar = function(btc){
    var dollar_one_btc = 8938.65;
    return dollar_one_btc * btc;
}

var d = 14776367535688;
var t = 84600;
var h = 2367890000;
console.log(hr_to_btc(h, d));